substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/JfrGCHeapSummaryEventSupport.java

```diff
@@ -0,0 +1,138 @@
+            JfrNativeEventWriterData data = StackValue.get(JfrNativeEventWriterData.class);
+            JfrNativeEventWriterDataAccess.initializeThreadLocalNativeBuffer(data);
+        
+            JfrNativeEventWriter.beginSmallEvent(data, JfrEvent.GCHeapSummary);
+        
+            JfrNativeEventWriter.putLong(data, start);  //   @Label("Start Time") @Timestamp("TICKS") long startTime;
+            
+            JfrNativeEventWriter.putLong(data, gcEpoch.rawValue()); //  @Label("GC Identifier") int gcId;
+            JfrNativeEventWriter.putLong(data, before.getId()); //  @Label("When") String when;
+        
+            // VirtualSpace
+            JfrNativeEventWriter.putLong(data, 0L);     // start
+            JfrNativeEventWriter.putLong(data, 0L);     // committedEnd : ulong
+            JfrNativeEventWriter.putLong(data, 0L);     // committedSize : ulong
+            JfrNativeEventWriter.putLong(data, 0L);     // reservedEnd : ulong
+            JfrNativeEventWriter.putLong(data, 0L);     // reservedSize : ulong
+        
+            JfrNativeEventWriter.putLong(data, heapUsed); //  @Unsigned @DataAmount("BYTES")  @Label("Heap Used") @Description("Bytes allocated by objects in the heap")  long heapUsed;
+        
+            JfrNativeEventWriter.endSmallEvent(data);
+        }
+
+    }
+
+    @Uninterruptible(reason = "Accesses a JFR buffer.")
+    public void emitGCHeapSummaryEventAfterGC(UnsignedWord gcEpoch, long start, long heapUsed) {
+        
+        if (JfrEvent.GCHeapSummary.shouldEmit()) {
+        
+            JfrNativeEventWriterData data = StackValue.get(JfrNativeEventWriterData.class);
+            JfrNativeEventWriterDataAccess.initializeThreadLocalNativeBuffer(data);
+        
+            JfrNativeEventWriter.beginSmallEvent(data, JfrEvent.GCHeapSummary);
+        
+            JfrNativeEventWriter.putLong(data, start); //   @Label("Start Time") @Timestamp("TICKS") long startTime;
+        
+            JfrNativeEventWriter.putLong(data, gcEpoch.rawValue()); //  @Label("GC Identifier") int gcId;
+            JfrNativeEventWriter.putLong(data, after.getId()); //  @Label("When") String when;
+        
+            // VirtualSpace           
+            JfrNativeEventWriter.putLong(data, 0L);     // start
+            JfrNativeEventWriter.putLong(data, 0L);     // committedEnd : ulong
+            JfrNativeEventWriter.putLong(data, 0L);     // committedSize : ulong
+            JfrNativeEventWriter.putLong(data, 0L);     // reservedEnd : ulong
+            JfrNativeEventWriter.putLong(data, 0L);     // reservedSize : ulong
+        
+            JfrNativeEventWriter.putLong(data, heapUsed); //  @Unsigned @DataAmount("BYTES")  @Label("Heap Used") @Description("Bytes allocated by objects in the heap")  long heapUsed;
+        
+            JfrNativeEventWriter.endSmallEvent(data);
+        }
+        
+    }
+}
+
+
+@AutomaticallyRegisteredFeature
+class JfrGCHeapSummaryEventFeature implements InternalFeature {
```

Comment:
Hmm I think this should actually be similar to `JfrGCEventFeature` and only be included if the default SerialGC is being used. See `isInConfiguration`

Reply:
I added isInConfiguration.